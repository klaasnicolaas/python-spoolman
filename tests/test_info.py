"""Test the Info model for Spoolman."""

from __future__ import annotations

from typing import TYPE_CHECKING

import pytest
from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from spoolman import Spoolman


async def test_get_info(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    spoolman_client: Spoolman,
) -> None:
    """Test getting the info."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/info",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("info.json"),
        ),
    )
    info = await spoolman_client.info()
    assert info == snapshot


@pytest.mark.parametrize(
    ("payload", "expected"),
    [
        ('{"status": "healthy"}', True),
        ('{"status": "unhealthy"}', False),
    ],
)
async def test_get_health(
    aresponses: ResponsesMockServer,
    spoolman_client: Spoolman,
    payload: str,
    expected: bool,  # noqa: FBT001
) -> None:
    """Test getting the health."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/health",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=payload,
        ),
    )
    health = await spoolman_client.health()
    assert health is expected
