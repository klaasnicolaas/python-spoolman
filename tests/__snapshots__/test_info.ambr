# serializer version: 1
# name: test_get_info
  Info(version='0.21.0', debug_mode=False, automatic_backups=True, db_type='sqlite', git_commit='053b20f', build_date=datetime.datetime(2024, 12, 1, 13, 55, 36, tzinfo=datetime.timezone.utc), data_dir='/root/.local/share/spoolman', logs_dir='/root/.local/share/spoolman', backups_dir='/root/.local/share/spoolman/backups')
# ---
